vcr/vcr.go

```diff
@@ -153,3 +255,59 @@ func (c *vcr) convert(query concept.Query) map[string]leia.Query {
 
 	return qs
 }
+
+func (c *vcr) resolveAssertionKey(id did.DID) (did.URI, error) {
+	doc, _, err := c.docResolver.Resolve(id, nil)
+	if err != nil {
+		return did.URI{}, err
+	}
+
+	keys := doc.AssertionMethod
+	for _, key := range keys {
+		kid := key.ID.String()
+		if c.keystore.PrivateKeyExists(kid) {
+			u, _ := url.Parse(kid)
+			return did.URI{URL: *u}, nil
+		}
+	}
+
+	return did.URI{}, fmt.Errorf("no valid assertion keys found for: %s", id.String())
+}
+
+func (c *vcr) generateProof(vc *did.VerifiableCredential, kid did.URI) error {
+	payload, err := json.Marshal(vc)
+	if err != nil {
+		return err
+	}
+
+	// create proof
+	pr := did.Proof{
+		Type:               "JsonWebSignature2020",
```

Comment:
define in `go-did/spec-registries.go`

Reply:
https://github.com/nuts-foundation/go-did/pull/25